shipyard_utils/files.py

```diff
@@ -158,31 +175,62 @@ def decompress_file(source_full_path, destination_full_path, compression):
     """
 
     if compression == 'zip':
-        with ZipFile(source_full_path, 'r') as zip:
-            zip.extractall(destination_full_path)
-        print(
-            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+        decompress_with_zip(
+            source_full_path,
+            destination_full_path,
+            compression)
 
     if compression == 'tar.bz2':
-        file = tarfile.open(source_full_path, 'r:bz2')
-        file.extractall(path=destination_full_path)
-        print(
-            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+        decompress_with_tar(
+            source_full_path,
+            destination_full_path,
+            compression)
 
     if compression == 'tar':
-        file = tarfile.open(source_full_path, 'r')
-        file.extractall(path=destination_full_path)
-        print(
-            f'Successfully extracted files from {source_full_path} to {destination_full_path}')
+        decompress_with_tar(
+            source_full_path,
+            destination_full_path,
+            compression)
 
     if compression == 'tar.gz':
-        file = tarfile.open(source_full_path, 'r:gz')
-        file.extractall(path=destination_full_path)
+        decompress_with_tar(
+            source_full_path,
+            destination_full_path,
+            compression)
```

Comment:
You refactor this the same way that you did the compress_with_tar. All three of the tar if clauses are the exact same. If compression starts with tar, then call decompress_with_tar.